test/theme.go

```diff
@@ -118,24 +130,50 @@ func Theme() fyne.Theme {
 				theme.SizeNameInputRadius:        float32(4),
 				theme.SizeNameSelectionRadius:    float32(4),
 			},
-		}
+		)
 	}
 	return defaultTheme
 }
 
+func newConfigurableTheme(name string, colors map[fyne.ThemeColorName]color.Color, fonts map[fyne.TextStyle]fyne.Resource, sizes map[fyne.ThemeSizeName]float32) fyne.Theme {
+	seen := map[color.Color]fyne.ThemeColorName{}
+	for cn, c := range colors {
+		if seen[c] != "" {
+			// This panic will never happen on released code because it will happen on CI runs prior release.
+			panic(fmt.Sprintf("duplicate color value %#v for color %s already used for color %s in theme %s", c, cn, seen[c], name))
+		}
+		seen[c] = cn
+	}
```

Comment:
Wouldn't it be wiser to run this in a test for our theme instead of doing it for each new theme created? We could even have some sort of helper to check that there are no duplicate colours.

Reply:
I can see your point. With a test one have to be aware that a third theme method (like `UglyAndPrettyTheme()`) would need to be checked but I think this is okay. With the current approach, a third method also might bypass the constructor, so there is no real advantage in my approach.
Will implement the tests.